### PR TITLE
Closes certificate file automatically during integration tests and im…

### PR DESCRIPTION
**Pull request by geircognite**
…proves docs listing the certificate auth mechanism as an example and removing the API key example. Also fixes mypy typing issue with assets.

## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by erlendvollset — APPROVED** [review]
Nice! 💯

File: cognite/client/data_classes/assets.py [hunk 1]
```diff
@@ -638,14 +638,14 @@ def parent_sort_fn(asset: Asset) -> str:
         )
         return mapping
 
-    def count_subtree(self, mapping: Dict[Optional[str], List[Asset]]) -> Dict[str, int]:
+    def count_subtree(self, mapping: Dict[Optional[str], List[Asset]]) -> Dict[Optional[str], int]:
```
Comment (haakonvt) [inline cognite/client/data_classes/assets.py:641]: Please revert type annotation changes for `count_subtree`. Check line 666, the mapping is purely str->str and hopefully mypy will understand that in a future version. Use —no-verify to skip pre-commit if you’re not on 3.8 😊

**Review by haakonvt — CHANGES_REQUESTED** [review]

File: cognite/client/data_classes/assets.py (hunk 1 above)
Comment (haakonvt) [reply cognite/client/data_classes/assets.py:641]: str->int * 😅

Comment (geircognite) [reply cognite/client/data_classes/assets.py]: Done. Sorry for the late response. Was a crazy week :)

**Review by haakonvt — APPROVED** [review]
## 🚀

**Pull request merged**